### PR TITLE
Add project accessions fields (#43)

### PR DESCRIPTION
**Pull request by danielsotirhos**

### CODE REVIEW

File: src/humancellatlas/data/metadata/api.py [hunk 1]
```diff
@@ -155,6 +159,10 @@ def __init__(self, json: JSON) -> None:
         self.publications = set(ProjectPublication.from_json(publication)
                                 for publication in content.get('publications', []))
         self.contributors = {ProjectContact.from_json(contributor) for contributor in content['contributors']}
+        self.insdc_project_accessions = {acc for acc in content.get('insdc_project_accessions', []) if acc}
```
Comment (hannes-ucsc) [inline src/humancellatlas/data/metadata/api.py]: Why do we filter empty and `null` accessions? If they occur in the input the API should reflect that.

Comment (danielsotirhos) [reply src/humancellatlas/data/metadata/api.py]: Ok, I see how if .get() returns [''] or [] the result would be an empty set for both cases.
I will remove the "if acc" which will cause these cases to result in {''} or {} respectively.

File: src/humancellatlas/data/metadata/api.py [hunk 2]
```diff
@@ -155,6 +159,10 @@ def __init__(self, json: JSON) -> None:
         self.publications = set(ProjectPublication.from_json(publication)
                                 for publication in content.get('publications', []))
         self.contributors = {ProjectContact.from_json(contributor) for contributor in content['contributors']}
+        self.insdc_project_accessions = {acc for acc in content.get('insdc_project_accessions', [])}
```
Comment (hannes-ucsc) [inline src/humancellatlas/data/metadata/api.py]: `{x for x in l}` is the same as `set(l)`

File: test/test.py [hunk 3]
```diff
@@ -251,6 +268,15 @@ def _assert_bundle(self, uuid, version, manifest, metadata_files,
         # noinspection PyDeprecation
         self.assertEqual(project.project_short_name, project.project_shortname)
 
+        if insdc_project_accessions != set() or project.insdc_project_accessions != set():
```
Comment (hannes-ucsc) [inline test/test.py]: I thought I had left a comment here in my previous review, but GH must have swallowed it.

The only way `insdc_project_accessions != set() or project.insdc_project_accessions != set()` can be false is if `insdc_project_accessions == set() and project.insdc_project_accessions == set()` but in that case the assertion would pass so there is no point in making it conditional.

File: src/humancellatlas/data/metadata/api.py (hunk 2 above)
Comment (danielsotirhos) [reply src/humancellatlas/data/metadata/api.py]: Ah yes, without a condition or modification of the values there is no need for the set comprehension. I will simplify these statements.

File: test/test.py (hunk 3 above)
Comment (danielsotirhos) [reply test/test.py]: Good catch. The condition started out only checking `insdc_project_accessions != set()` but when `project.insdc_project_accessions != set()` was added the condition became superfluous. Will be corrected in next commit.

**Pull request merged**